R/calcOutputWrapper.R

```diff
@@ -0,0 +1,115 @@
+#' disagreements will be reported via a warning.
+#' \item \bold{aggregationFunction} (optional | default = toolAggregate) - Function to be used to
+#' aggregate data from country to regions. The function must have the argument \code{x} for
+#' the data itself and \code{rel} for the relation mapping between countries and regions and
+#' must return the data as magpie object in the spatial resolution as defined in rel.
+#' \item \bold{aggregationArguments} (optional) - List of additional, named arguments to be supplied
+#' to the aggregation function. In addition to the arguments set here, the function will be
+#' supplied with the arguments \code{x}, \code{rel} and if provided/deviating from the default
+#' also \code{weight} and \code{mixed_aggregation}.
+#' \item \bold{putInPUC} (optional) boolean which decides whether this calculation should be added to a puc file
+#' which contains non-aggregated data and can be used to later on aggregate the data to resolutions of own choice.
+#' If not set \code{calcOutput} will try to determine automatically, whether a file is being required for the puc file
+#' or not, but in more complex cases (e.g. if calculations below top-level have to be run as well) this setting can
+#' be used to manually tweak the puc file list. CAUTION: Incorrect settings will cause corrupt puc files,
+#' so use this setting with extreme care and only if necessary.
+#' \item \bold{cache} (optional) boolean which decides whether a cache file should be written (if caching is active)
+#' or not. Default setting is TRUE. This can be for instance useful, if the calculation itself is quick, but
+#' the corresponding file sizes are huge. Or if the caching for the given data type does not support storage
+#' in RDS format. CAUTION: Deactivating caching for a data set which should be part of a PUC file
+#' will corrupt the PUC file. Use with care.
+#' }
+#' @author Jan Philipp Dietrich
+#' @seealso \code{\link{setConfig}}, \code{\link{calcTauTotal}},
+#' @examples
+#' \dontrun{
+#'
+#' a <- calcOutput(type = "TauTotal")
+#' }
+#'
+#' @importFrom magclass nyears nregions getComment<- getComment getYears clean_magpie write.report write.magpie
+#' getCells getYears<- is.magpie dimSums
+#' @importFrom utils packageDescription read.csv2 read.csv
+#' @importFrom digest digest
+#' @importFrom withr defer local_dir
+#' @export
+
+calcOutputWrapper <- function(
+    calc, aggregate = TRUE, file = NULL, years = NULL, round = NULL, signif = NULL,
+    supplementary = FALSE, append = FALSE, warnNA = TRUE, na_warning = NULL, try = FALSE,
+    regionmapping = NULL, writeArgs = NULL, ...) {
+                       
+    type <- sub("^.*?calc", "", deparse(substitute(calc)))
+    calcOutput(
+        type, aggregate=aggregate, file=file, years=years, round=round, signif=signif,
+        supplementary=supplementary, append=append, warnNA=warnNA, na_warning=na_warning, try=try,
+        regionmapping=regionmapping, writeArgs=writeArgs, ...=...
```

Comment:
```suggestion
        regionmapping=regionmapping, writeArgs=writeArgs, ...
```
I'm pretty sure the other way won't work, or did you try it?

Reply:
Didn't try it, I don't know any usecase for it... I have changed it now.